doc/source/release/1.15.0-notes.rst

```diff
@@ -17,96 +17,1431 @@ run your code with ``python -Wd`` and check for ``DeprecationWarning`` s).
+  vector-in, vector-out functions.
+* Use `scipy.differentiate.hessian` for second-order partial derivatives of
+  vector-in, scalar-out functions.
+
+All functions use high-order finite difference rules with adaptive (real)
+step size. To facilitate batch computation, these functions are vectorized
+and support several Array API compatible array libraries in addition to NumPy
+(see "Array API Standard Support" below).
+
+`scipy.integrate` improvements
+==============================
+- The ``QUADPACK`` Fortran77 package has been ported to C.
+- `scipy.integrate.lebedev_rule` computes abscissae and weights for
+  integration over the surface of a sphere.
+- `scipy.integrate.nsum` evaluates finite and infinite series and their
+  logarithms. To facilitate batch computation, the function is vectorized and
+  supports several Array API compatible array libraries in addition to
+  NumPy (see "Array API Standard Support" below).
+- `scipy.integrate.tanhsinh` is now exposed for public use, allowing
+  evaluation of a convergent integral using tanh-sinh quadrature.
+- The new `scipy.integrate.cubature` function supports multidimensional
+  integration, and has support for approximating integrals with
+  one or more sets of infinite limits.
 
 
 `scipy.interpolate` improvements
 ================================
+- `scipy.interpolate.AAA` adds the AAA algorithm for barycentric rational
+  approximation of real or complex functions.
+- `scipy.interpolate.FloaterHormannInterpolator` adds barycentric rational
+  interpolation.
+- New functions `scipy.interpolate.make_splrep` and
+  `scipy.interpolate.make_splprep` implement construction of smoothing splines.
+  The algorithmic content is equivalent to FITPACK (``splrep`` and ``splprep``
+  functions, and ``*UnivariateSpline`` classes) and the user API is consistent
+  with ``make_interp_spline``: these functions receive data arrays and return
+  a `scipy.interpolate.BSpline` instance.
+- New generator function `scipy.interpolate.generate_knots` implements the
+  FITPACK strategy for selecting knots of a smoothing spline given the
+  smoothness parameter, ``s``. The function exposes the internal logic of knot
+  selection that ``splrep`` and ``*UnivariateSpline`` was using.
 
 
 `scipy.linalg` improvements
 ===========================
+- `scipy.linalg.interpolative` Fortran77 code has been ported to Cython.
+- `scipy.linalg.solve` supports several new values for the ``assume_a``
+  argument, enabling faster computation for diagonal, tri-diagonal, banded, and
+  triangular matrices. Also, when ``assume_a`` is left unspecified, the
+  function now automatically detects and exploits diagonal, tri-diagonal,
+  and triangular structures.
+- `scipy.linalg` matrix creation functions (`scipy.linalg.circulant`,
+  `scipy.linalg.companion`, `scipy.linalg.convolution_matrix`,
+  `scipy.linalg.fiedler`, `scipy.linalg.fiedler_companion`,
+  `scipy.linalg.leslie`, and `scipy.linalg.toeplitz`) now support batch
```

Comment:
`toeplitz` doesn't support batching (yet): it currently ravels >1-d input, which has been deprecated, so it can support batching in 1.17

Reply:
https://github.com/scipy/scipy/pull/21446/files#diff-6482beada610f60c724811cf77561f2bde8fd44ed2349939e505d28746a6d078R38-R42

Reply:
Oops. Good catch.